File: 150-evaluate-reverse-polish-notation/150-evaluate-reverse-polish-notation.ts
function evalRPN(tokens: string[]): number {
    
    let stack: string[] = []
    
    let operator = ["+","-","*","/"]
    
    
    for(let token of tokens){
        stack.push(token)
        
        
        if(operator.includes(token)){
            let operator = stack.pop()
            let num2 = parseInt(stack.pop())
            let num1 = parseInt(stack.pop())
            
            
            if(operator == '+'){
                stack.push((num1 + num2).toString())
            }
            if(operator == '-'){
                stack.push((num1 - num2).toString())
            }
            if(operator == '*'){
                stack.push((num1 * num2).toString())
            }
            if(operator == '/'){
                stack.push((num1 / num2).toString())
            }
        }
    }
    
    return parseInt(stack.pop())

    
    
};

File: 150-evaluate-reverse-polish-notation/150-evaluate-reverse-polish-notation.java
class Solution {
    public int evalRPN(String[] tokens) {
   
    Stack<String> stack = new Stack<String>();
    
    String operators = "+-*/";
    
    
    for(String token : tokens){
        
        if(!operators.contains(token))
        
            stack.push(token);
        
        
        else{
            
            int num2 = Integer.valueOf(stack.pop());
            int num1 = Integer.valueOf(stack.pop());
            
            
            if(token.equals("+")){
                stack.push(String.valueOf(num1 + num2));
            }
            if(token.equals("-")){
                stack.push(String.valueOf(num1 - num2));
            }
            if(token.equals("*")){
                stack.push(String.valueOf(num1 * num2));
            }
            if(token.equals("/")){
                stack.push(String.valueOf(num1 / num2));
            }
            
        }
    }
    
    return Integer.valueOf(stack.pop());

    }
    
}